lib/src/connector/ischool_plus_connector.dart

```diff
@@ -36,79 +36,67 @@ class ISchoolPlusConnector {
+      final ssoIndexTagNode = html.parse(ssoIndexResponse);
+      final ssoIndexNodes = ssoIndexTagNode.getElementsByTagName("input");
+      final ssoIndexJumpUrl = ssoIndexTagNode.getElementsByTagName("form")[0].attributes["action"];
 
-      tagNode = html.parse(result.toString().trim());
-      nodes = tagNode.getElementsByTagName("input");
-      data.clear();
-      for (final node in nodes) {
+      final Map<String, String> oauthData = {};
+      for (final node in ssoIndexNodes) {
         final name = node.attributes['name'];
         final value = node.attributes['value'];
-        data[name] = value;
+        oauthData[name] = value;
       }
 
       // Step 2
-      // The `jumpUrl` should be "oauth2Server.do".
+      // The ssoIndexJumpUrl should be "oauth2Server.do".
       // If not, it means that the school server has changed.
-      // TODO: Add a validation measurement to check whether Step 1 is died or not. (It should not die if auth is correct)
-      final jumpUrl = tagNode.getElementsByTagName("form")[0].attributes["action"];
-      parameter = ConnectorParameter("${NTUTConnector.host}$jumpUrl");
-      parameter.data = data;
-
-      Response<dynamic> jumpResult = (await Connector.getDataByPostResponse(parameter));
-      tagNode = html.parse(jumpResult.data.toString().trim());
-      nodes = tagNode.getElementsByTagName("a");
-
-      // Step 3
-      // The redirectUrl is provided by <a> HTML DOM on Step 2.
-      // It should be https://istudy.ntut.edu.tw/login2.php with lot of the parameters.
-      final redirectUrl = nodes.first.attributes["href"];
-      parameter = ConnectorParameter(redirectUrl);
-      await Connector.getDataByGet(parameter);
-
-      // Perform retry for cryptic API errors (?).
-      // If the string `connect lost` be found in the response, we will do the retry.
-
-      // [2023-10-21] We may not need this since the step was changed.
-      // TODO: Remove I-School retry loop since it's outdated.
-
-      int retryTimes = 3;
-      do {
-        if (jumpResult.data.toString().contains('connect lost')) {
-          // Take a short delay to avoid being blocked.
-          await Future.delayed(const Duration(milliseconds: 100));
-          jumpResult = (await Connector.getDataByPostResponse(parameter));
+      // The response status code to this request should result in
+      // "302" (the page has moved to a new location), which triggers automatic redirection
+      // feature included in dio connector, thus no further actions needed.
+      for (int retry = 0; true; retry++) {
+        if (retry == 3) break;
+        final jumpParameter = ConnectorParameter("${NTUTConnector.host}$ssoIndexJumpUrl");
+        jumpParameter.data = oauthData;
+        final jumpResult = (await Connector.getDataByPostResponse(jumpParameter));
+        if (jumpResult.statusCode == 302) {
+          return ISchoolPlusConnectorStatus.loginSuccess;
         } else {
```

Comment:
This might can fix the login issue.
and the `else` in end is useless, or it's a code style i not known yet.
but that delay function can put out of `else` block and work same.

```suggestion
        final jumpResult = await Connector.getDataByPostResponse(jumpParameter);
        if (jumpResult.statusCode == 302) {
          final login2Parameter = ConnectorParameter( jumpResult.headers['location'] );
          await Connector.getDataByGet(login2Parameter);
          return ISchoolPlusConnectorStatus.loginSuccess;
        }
```

Reply:
[f69f2c3](https://github.com/NEO-TAT/tat_flutter/pull/225/commits/f69f2c3f01c7f41b44211654dfbab00d744ca266)